phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexToolForDeleteBeforeRebuildIT.java

```diff
@@ -0,0 +1,220 @@
+        Map<String, String> serverProps = Maps.newHashMapWithExpectedSize(2);
+        serverProps.put(QueryServices.STATS_GUIDEPOST_WIDTH_BYTES_ATTRIB, Long.toString(20));
+        serverProps.put(QueryServices.MAX_SERVER_METADATA_CACHE_TIME_TO_LIVE_MS_ATTRIB, Long.toString(5));
+        serverProps.put(QueryServices.EXTRA_JDBC_ARGUMENTS_ATTRIB,
+            QueryServicesOptions.DEFAULT_EXTRA_JDBC_ARGUMENTS);
+        Map<String, String> clientProps = Maps.newHashMapWithExpectedSize(2);
+        clientProps.put(QueryServices.USE_STATS_FOR_PARALLELIZATION, Boolean.toString(true));
+        clientProps.put(QueryServices.STATS_UPDATE_FREQ_MS_ATTRIB, Long.toString(5));
+        clientProps.put(QueryServices.TRANSACTIONS_ENABLED, Boolean.TRUE.toString());
+        clientProps.put(QueryServices.FORCE_ROW_KEY_ORDER_ATTRIB, Boolean.TRUE.toString());
+        setUpTestDriver(new ReadOnlyProps(serverProps.entrySet().iterator()),
+            new ReadOnlyProps(clientProps.entrySet().iterator()));
+    }
+
+    @Before public void prepareTest() throws Exception{
+        Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        conn = DriverManager.getConnection(getUrl(), props);
+
+        schemaName = generateUniqueName();
+        dataTableName = generateUniqueName();
+        viewName = generateUniqueName();
+        dataTableFullName = SchemaUtil.getTableName(schemaName, dataTableName);
+        viewFullName = SchemaUtil.getTableName(schemaName, viewName);
+        globalIndexName = generateUniqueName();
+        globalIndexFullName = SchemaUtil.getTableName(schemaName, globalIndexName);
+        createTestTable(getUrl(), String.format(DATA_TABLE_DDL, dataTableFullName));
+        createTestTable(getUrl(), String.format(VIEW_DDL, viewFullName, dataTableFullName));
+        String dataTableUpsert = String.format(UPSERT_SQL, dataTableFullName);
+        PreparedStatement stmt = conn.prepareStatement(dataTableUpsert);
+        upsertRow(stmt, "tenantID1",1, "name1", 9991);
+        upsertRow(stmt, "tenantID1",2, "name2", 9992);
+        upsertRow(stmt, "tenantID1",3, "name3", 9993);
+        conn.commit();
+    }
+
+    private void upsertRow(PreparedStatement stmt, String tenantId, int id, String name, int zip)
+            throws SQLException {
+        int index = 1;
+        stmt.setString(index++, tenantId);
+        stmt.setInt(index++, id);
+        stmt.setString(index++, name);
+        stmt.setInt(index++, zip);
+        stmt.executeUpdate();
+    }
+
+    @After public void teardown() throws SQLException {
+        if (conn != null) {
+            conn.close();
+        }
+    }
+
+    @Test
+    public void testDeleteBeforeRebuildForViewIndex() throws Exception {
+        String createViewIndex = String.format(INDEX_GLOBAL_DDL, globalIndexName, viewFullName);
+        PreparedStatement stmt = conn.prepareStatement(createViewIndex);
+        stmt.execute();
+        runIndexTool(schemaName, viewName, globalIndexName, -1);
```

Comment:
Could you please refactor the test name to convey that its a failure scenario and same for another failure scenario? or add a comment with description for each test?

Reply:
I added ShouldFail